### ISSUE TITLE
Cross-dependency version not updated during publish

### ISSUE THREAD

**Issue by gigabo** [kittens/lerna#50]
Hi there.  Just getting started with lerna.  Very cool.

I've experienced unexpected behavior.  Not sure yet if I did something wrong or if this was a bug.

I have a [project](https://github.com/redfin/react-server) with two packages.  The [`react-server`](https://github.com/redfin/react-server/blob/master/packages/react-server/package.json#L69) package depends on [`react-server-gulp-module-tagger`](https://github.com/redfin/react-server/blob/master/packages/react-server-gulp-module-tagger/package.json).

I've just [published `v0.0.6`](https://github.com/redfin/react-server/commit/a3a40d6877e075527f3c069923dd7ecec06f24f6) which introduces this dependency.

What I thought would happen: The `lerna publish` updates the version in `react-server`'s `devDependencies` to point to the newly published version.

What actually happened: The `devDependency` version was not modified.  This [causes `lerna bootstrap` to fail](https://travis-ci.org/redfin/react-server/builds/108697795#L665) in travis ci and breaks our build.  Strangely `lerna bootstrap` still works fine _locally_.

I'm confused.  Have I set up the cross-dependency incorrectly?  Has something gone wrong during publish?

Thanks!

Bo

**Comment by thejameskyle** [comment]
Related: https://github.com/kittens/lerna/issues/27

**Comment by dtothefp** [comment]
I have several packages that are linked/published by `lerna` https://github.com/dtothefp/build-boiler

I feel like I'm having a similar problem that happens on every travis build. I run `lerna bootstrap` on travis and it seems to complete successfully 

https://travis-ci.org/dtothefp/build-boiler/builds/116948869

<img width="1062" alt="screen shot 2016-03-18 at 1 45 22 pm" src="https://cloud.githubusercontent.com/assets/4656726/13886956/d666a66a-ed0f-11e5-842d-603d578c0026.png">
But upon running the build task in my module which depends upon some of my linked `packages` it seems the the `npm link` has failed

```
Error: Cannot find module '/home/travis/build/dtothefp/build-boiler/packages/boiler-utils'
```

**Comment by thejameskyle** [comment]
This sounds like a separate issue

**Comment by dtothefp** [comment]
@thejameskyle ok I made a new issue here https://github.com/kittens/lerna/issues/87. Seems like pretty common functionality to want to be able to run `lerna` builds and potentially `publish` in some sort of CI environment like Travis

**Comment by thejameskyle** [comment]
Fixed in 2.0.0-beta.4
